lib/upperkut/processor.rb

```diff
@@ -13,10 +13,11 @@ def process
       items = @worker.fetch_items.freeze
 
       @worker.server_middlewares.invoke(@worker, items) do
-        @worker_instance.perform(items.map(&:dup))
+        @worker_instance.perform(items)
```

Comment:
@nandosousafr I had to remove `.dup` to keep the same reference in order to check if an item was accepted by the worker.